src/components/pages/donation_form/DonorSummary.vue

```diff
@@ -0,0 +1,88 @@
+<template>
+	<div class="donor-summary" v-if="hasData">
+		<h4>{{ $t('donation_summary_contact_data_header') }}</h4>
+		<strong v-if="name">{{ name }}<br/></strong>
+		<template v-if="cleanedStreetAddress">{{ cleanedStreetAddress }}<br/></template>
+		<template v-if="postcodeCity">{{ postcodeCity }}<br/></template>
+		<template v-if="country">{{ country }}<br/></template>
+		<template v-if="email">{{ email }}<br/></template>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue';
+import { clearStreetAndBuildingNumberSeparator } from '@src/util/street_and_building_number_tools';
+import type { Country } from '@src/view_models/Country';
+import type { Salutation } from '@src/view_models/Salutation';
+import type { Address } from '@src/view_models/Address';
+import { useI18n } from 'vue-i18n';
+
+interface Props {
+	address: Address;
+	countries: Array<Country>;
+	salutations: Array<Salutation>;
+}
+const props = defineProps<Props>();
+
+const { t } = useI18n();
+
+const name = computed( () => {
+	if ( props.address.companyName ) {
+		return props.address.companyName;
+	}
+
+	const nameParts: string[] = [];
+
+	if ( props.address.salutation || props.address.title ) {
+		const localisationKey = props.address.title
+			? 'address_salutation_academic_title'
+			: 'address_salutation_no_academic_title';
+
+		const renderedSalutationOrTitle = t( localisationKey, {
+			salutation: props.address.salutation || '',
+			title: props.address.title || '',
+		} );
+
+		if ( renderedSalutationOrTitle.trim() ) {
+			nameParts.push( renderedSalutationOrTitle.trim() );
+		}
+	}
+
+	if ( props.address.firstName ) {
+		nameParts.push( props.address.firstName );
+	}
+
+	if ( props.address.lastName ) {
+		nameParts.push( props.address.lastName );
+	}
+
+	return nameParts.join( ' ' );
+} );
+
+const cleanedStreetAddress = computed( () =>
+	clearStreetAndBuildingNumberSeparator( props.address.street || '' )
+);
+
+const postcodeCity = computed( () =>
+	[ props.address.postcode, props.address.city ]
+		.filter( Boolean )
+		.join( ' ' )
+);
+
+const country = computed( () => {
+	const countryObject = props.countries.find(
+		( c ) =>
+			c.countryCode === props.address.country
+	);
+	return countryObject ? countryObject.countryFullName : '';
+} );
+
+const email = computed( () => props.address.email || '' );
+
+const hasData = computed( () =>
+	Boolean( name.value ) ||
```

Comment:
These evaluations should probably be `&&` instead of `||`, as it will hide the summary until all the required data is entered. The summary looks weird when there's only some of the data filled out.
 
![Screenshot 2025-06-03 at 08 52 02](https://github.com/user-attachments/assets/71d12bf6-f43f-449d-86a7-f1995ea9c812)